validphys2/src/validphys/convolution.py

```diff
@@ -112,9 +112,15 @@ def _predictions(dataset, pdf, fkfunc):
             "commondata and is not supported."
         )
     cuts = dataset.cuts.load()
-    all_predictions = [
-        fkfunc(load_fktable(fk).with_cuts(cuts), pdf) for fk in dataset.fkspecs
-    ]
+    all_predictions = []
+    for fk in dataset.fkspecs:
+        fk_w_cuts = load_fktable(fk).with_cuts(cuts)
+        all_predictions.append(fkfunc(fk_w_cuts, pdf))
```

Comment:
Taste. In general I don't like unnecessary list comprehension because it makes the code harder to read (for me).

I usually don't change it when someone already wrote it like that but in this particular case loading the fktable in one step and then computing the convolution facilitates debugging (of which I had to do quite a bit to ensure this was working as it should).